AK/StringUtils.cpp

```diff
@@ -427,6 +427,19 @@ String to_titlecase(StringView const& str)
     return builder.to_string();
 }
 
+size_t count(StringView const& str, StringView const& needle)
+{
+    if (needle.is_empty())
+        return str.length();
+
+    size_t count = 0;
+    for (size_t i = 0; i < str.length() - needle.length() + 1; ++i) {
+        if (str.substring_view(i).starts_with(needle))
+            count++;
+    }
+    return count;
+}
```

Comment:
I'm sorta leaning towards using KMP here instead, by producing the table once and successively attempting to match it, counting the number of successes, something like:
```c++
{
    if (needle.is_empty())
        return str.length();

    auto table = prepare_kmp_partial_table(needle);
    size_t count = 0;
    auto it = str.begin();
    auto old_it = it;
    while (it != str.end()) {
        old_it = it;
        auto maybe_index = memmem_kmp(it, str.end(), table);
        if (maybe_index.has_value())
            count++;
        if (it == old_it) ++it;
    }
    return count;
}
```

we already have an implementation of kmp for memmem, so splitting it up should be reasonable, theoretically this takes the complexity from O(n*m) to O(n+m), though I'm not sure if we're calling `count(StringView)` with long enough strings enough times to make this matter at all...

Reply:
I think the absolute worst case for kmp here would be equivalent to the previous algorithm, O(m) for the table generation (equivalent to strlen on the needle), and O(n) for the string search (strstr() on the subject, provided that the needle shows up only at the end).

So we definitely won't have any perf degradation here (unless I'm missing something?)

Reply:
Right, I'm not talking about the O(x), I'm talking about real world perf degrading due to the higher "hidden constant", i.e. the a larger amount of branches and the generally higher cyclomatic complexity, which is made trivial by a large enough string, but not necessarily so for small ones

Reply:
Fair enough I suppose.
either way, that's not gonna stand in the way of this PR.

Reply:
I'll add a FIXME to benchmark this vs KMP and switch to it if it's better.